### PR TITLE
Ft gh 675 flacon liveness fix

### PR DESCRIPTION
**Pull request by Fab-T**
### Purpose

- https://broadinstitute.atlassian.net/browse/GH-675

### Changes

- modified handler adding new function report status
- modified run to disable cache (even though not working great)
- modified routes checking for html file generated by report status

### Review Instructions

- Instructions to verify 
Falcon service should be running and Liveness Probe shouldn't restart container
Logs : container logs should show new line for handler stating report status is successful

Interacting with container : 
Check that file exist and timestamp updated every minute or so
> kubectl exec falcon-xxIDxx -c falcon -- cat falcon/templates/handler_status.html

Expected result:
<html>
<head></head>
<body><p>Time when report my status was generated: 06-Mar-2020 (21:53:09)</p></body>
</html>

Check monitoring status function by curling /health
> kubectl exec falcon-xxIDxx -c falcon -- curl http://localhost:8000/health

Expected result:
Time when report my status was generated: 06-Mar-2020 (21:53:09)

Simulating failure:
By changing the parameter in settings.py you can simulate a failure
change maxDelay=5 > maxDelay=1 to reduce to 1 min the delay accepted between when was the file last modified (handler_status.html)
Change 
settings['queue_update_interval'] = int(settings.get('queue_update_interval', 1))
to 
settings['queue_update_interval'] = int(settings.get('queue_update_interval', 10))
Sleep time between report_status call that should modified the html file.

Update container
kubectl cp settings.py falcon-xxIDxx:falcon -c falcon

Restart Gunicorn : 
kubectl exec falcon-6977f47fd5-xsqwz -c falcon -- ps -ef

Expected result:
UID          PID    PPID  C STIME TTY          TIME CMD
root           1       0  0 17:24 ?        00:00:02 /usr/local/bin/python /usr/local/bin/gunicorn falcon.run:app -b 0.0.0.0:8000
root         **157**       1  0 17:26 ?

Kill the PID of the worker: 157
kubectl exec falcon-6977f47fd5-xsqwz -c falcon -- kill -HUP **157**

This should restart gunicorn with new config...
Check again for health:
kubectl exec falcon-xxIDxx -c falcon -- curl http://localhost:8000/health

Wait a minute and check again
kubectl exec falcon-xxIDxx -c falcon -- curl http://localhost:8000/health

Expected Result:
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>**reached max delay**

Liveness probe should detect the Error 500 as well and restart the container.
If not wait a bit longer for the container to restart...

### CODE REVIEW

**Comment by Fab-T** [comment]
Not sure why after I installed locally and ran the exact same command as the travis build I am still failing here !

File: .gitignore [hunk 1]
```diff
@@ -137,3 +137,6 @@ Icon
 Network Trash Folder
 Temporary Items
 .apdisk
+
+##JetBrain Stuff
+.idea
```
Comment (rexwangcc) [inline .gitignore]: nit: new line at the end

File: Dockerfile [hunk 2]
```diff
@@ -7,6 +7,6 @@ COPY requirements.txt .
 
 RUN pip install -r requirements.txt
 
-COPY . .
+COPY falcon falcon
```
Comment (rexwangcc) [inline Dockerfile]: like this!

File: falcon/queue_handler.py [hunk 3]
```diff
@@ -117,7 +150,8 @@ def execution_event(self):
             self.set_queue(self.create_empty_queue(-1))
 
             self.enqueue(workflows)
-        else:
+        else :
```
Comment (rexwangcc) [inline falcon/queue_handler.py]: nit: no need to have whitespace between `else` and `:`

File: falcon/routes.py [hunk 4]
```diff
@@ -1,20 +1,33 @@
...
+        if now - file_mod_time > max_delay:
+            abort(500,'reached max delay')
         else:
-            display_name = '{}-thread'.format(falcon_thread_name)
-            active_falcon_threads[display_name] = str(thread.ident)
-    return json.dumps(active_falcon_threads)
+            return render_template(docRootFile)
+
+    except Exception as exc:
+        abort(500,exc)
```
Comment (rexwangcc) [inline falcon/routes.py]: nit: new line at the end.

File: falcon/settings.py [hunk 5]
```diff
@@ -2,6 +2,10 @@
 import os
 from cromwell_tools.cromwell_auth import CromwellAuth
 
+#Global variables
+docRootPath='falcon/templates/'     #Path to Flask docRoot - in the repo
+docRootFile='handler_status.html'    #file to be returned to liveness Probe - in the repo
+maxDelay=5  # we consider if docRootFile is older than 5 mins then the Thread is Frozen
```
Comment (rexwangcc) [inline falcon/settings.py]: nit: use CAPS for the global vars, e.g. `MAX_DELAY=5`

File: .gitignore (hunk 1 above)
Comment (Fab-T) [reply .gitignore]: done

File: falcon/queue_handler.py (hunk 3 above)
Comment (Fab-T) [reply falcon/queue_handler.py]: done

File: Dockerfile (hunk 2 above)
Comment (Fab-T) [reply Dockerfile]: thanks

File: falcon/routes.py (hunk 4 above)
Comment (Fab-T) [reply falcon/routes.py]: done

File: falcon/settings.py (hunk 5 above)
Comment (Fab-T) [reply falcon/settings.py]: done

**Pull request merged**